### PR TITLE
improve testability

### PR DESCRIPTION
**Pull request by wegenmic**

### CODE REVIEW

File: gdk-core/src/main/java/com/github/libgraviton/gdk/GravitonApi.java [hunk 1]
```diff
@@ -106,6 +106,10 @@ public String getBaseUrl() {
         return baseUrl;
     }
 
+    public String getEndpointUrl() {
```
Comment (q-src) [inline gdk-core/src/main/java/com/github/libgraviton/gdk/GravitonApi.java]: This method seems to be unneeded.

File: gdk-core/src/main/java/com/github/libgraviton/gdk/api/GravitonRequest.java [hunk 2]
```diff
@@ -13,17 +13,20 @@
...
-    private final HeaderBag headers;
+    private HeaderBag headers;
 
-    private final String body;
+    private String body;
 
-    private final List<Part> parts;
+    private List<Part> parts;
+
+    protected GravitonRequest() {
```
Comment (q-src) [inline gdk-core/src/main/java/com/github/libgraviton/gdk/api/GravitonRequest.java]: Why do we need an empty constructor?

File: gdk-core/src/main/java/com/github/libgraviton/gdk/api/GravitonResponse.java [hunk 3]
```diff
@@ -32,6 +32,9 @@
 
     private ObjectMapper objectMapper;
 
+    protected GravitonResponse() {
```
Comment (q-src) [inline gdk-core/src/main/java/com/github/libgraviton/gdk/api/GravitonResponse.java]: Why do we need an empty constructor?

**Comment by wegenmic** [comment]
@q-src all those changes makes it easier for user of this library to test their code. This means It should be possible to do painless mocking. 
getEndpointUrl() can be used to mock this part of the class specifically
GravitonRequest and GravitonResponse need a default constructor so it's possible for Mockito to create mocks / partial mocks of these classes

**Pull request merged**